classes/note/NoteDAO.inc.php

```diff
@@ -288,16 +303,20 @@ public function deleteObject($note)
      */
     public function deleteById($noteId, $userId = null)
     {
-        $params = [(int) $noteId];
+        $submissionFileCollector = Repo::submissionFile()
+            ->getCollector()
+            ->filterByAssoc(Application::ASSOC_TYPE_NOTE, [$noteId]);
+
+        Repo::submissionFile()->deleteMany($submissionFileCollector);
+
+        $query = DB::table('notes')
```

Comment:
I think you forgot to implement the `$userId` variable here.

Reply:
Do you mean that

```
if ($userId) {
   $query->where('user_id', '=', $userId);
}
```

or something else?

Reply:
You're right, I missed it just below. I was looking for a fully fluent implementation of the query build:
```php
$query = DB::table('notes')
  ->where('note_id', '=', $noteId)
  ->when($userId !== null, function($query, $userId) {
    $query->where('user_id', '=', $userId);
  });
```